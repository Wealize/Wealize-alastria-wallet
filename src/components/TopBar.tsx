import React from 'react'

import { Colors } from '../utils/themes'
import { TopBarContainer, Title, TopBarView } from './styles/TopBar.styles'

interface TopBarProps {
  title: string
  textColor?: string
  color?: string
}

const TopBar = ({
  title,
  textColor = '#ffffff',
  color = Colors.principalDark
}: TopBarProps) => {
  return (
    <TopBarView>
      <TopBarContainer backgroundColor={color}>
        <Title textColor={textColor}>{title}</Title>
      </TopBarContainer>
    </TopBarView>
  )
}

export default TopBar
