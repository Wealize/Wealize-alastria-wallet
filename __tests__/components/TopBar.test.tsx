import React from 'react'
import renderer from 'react-test-renderer'

import TopBar from '../../src/components/TopBar'

describe('OnboardginComponent', () => {
  it('should render correctly TopBar with Props', () => {
    const result = renderer.create(
      <TopBar title="title" textColor="red" color="blue" />
    )
    expect(result).toMatchSnapshot()
  })

  it('should render correctly TopBar without Props', () => {
    const result = renderer.create(<TopBar title="title" />)
    expect(result).toMatchSnapshot()
  })
})
