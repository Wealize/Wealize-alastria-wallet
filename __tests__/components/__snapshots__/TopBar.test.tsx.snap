// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardginComponent should render correctly TopBar with Props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    className="sc-bdvvtL icfKFC"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  >
    <Text
      className="sc-gsDKAQ cYEvsf"
      style={
        Object {
          "color": "red",
        }
      }
    >
      title
    </Text>
  </View>
</View>
`;

exports[`OnboardginComponent should render correctly TopBar without Props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    className="sc-bdvvtL icfKFC"
    style={
      Object {
        "backgroundColor": "#002b6a",
      }
    }
  >
    <Text
      className="sc-gsDKAQ cYEvsf"
      style={
        Object {
          "color": "#ffffff",
        }
      }
    >
      title
    </Text>
  </View>
</View>
`;
